tests/main/dsl/test_caches.py

```diff
@@ -0,0 +1,179 @@
+        "test_caches_OrchestratedProgam___call__",
+    ) or os.path.exists(
+        f"{working_dir}/.gt_cache_FV3_A/dacecache/OrchestratedProgam___call__",
+    )
+
+    # Compile in another directory
+
+    custom_path = f"{working_dir}/.my_cache_path"
+    gt_config.cache_settings["root_path"] = custom_path
+    p1 = OrchestratedProgam(backend, DaCeOrchestration.BuildAndRun)
+    p1()
+    assert os.path.exists(
+        f"{custom_path}/.gt_cache_FV3_A/dacecache/"
+        "test_caches_OrchestratedProgam___call__",
+    ) or os.path.exists(
+        f"{working_dir}/.gt_cache_FV3_A/dacecache/OrchestratedProgam___call__",
+    )
+
+    # Check relocability by copying the second cache directory,
+    # changing the path of gt_config.cache_settings and trying to Run on it
+    relocated_path = f"{working_dir}/.my_relocated_cache_path"
+    shutil.copytree(custom_path, relocated_path, dirs_exist_ok=True)
+    gt_config.cache_settings["root_path"] = relocated_path
+    p2 = OrchestratedProgam(backend, DaCeOrchestration.Run)
+    p2()
+
+    # Generate a file exists error to check for bad path
+    bogus_path = "./nope/notatall/nothappening"
+    gt_config.cache_settings["root_path"] = bogus_path
+    with pytest.raises(RuntimeError):
+        OrchestratedProgam(backend, DaCeOrchestration.Run)
+
+    # Restore cache settings
+    gt_config.cache_settings["root_path"] = original_root_directory
+
+
+@pytest.mark.parametrize(
+    "backend",
+    [
+        pytest.param("gt:cpu_ifirst"),
+        pytest.param("dace:cpu"),
+    ],
+)
+def test_relocatability(backend):
+    # TODO: test work - but crashes when chained with other
+    #       see https://github.com/GEOS-ESM/pace/issues/16
+    pass
+    # import os
+    # import shutil
+
+    # working_dir = os.getcwd()
+
+    # # Compile on default
+    # p0 = OrchestratedProgam(backend, DaCeOrchestration.Python)
+    # p0()
+    # assert os.path.exists(
+    #     f"{working_dir}/.gt_cache_000000/py38_1013/gtcpu_ifirst/__main__/_stencil/"
+    # )
+
+    # # Compile in another directory
+    # from gt4py.cartesian import config as gt_config
+
+    # custom_path = f"{working_dir}/.my_cache_path"
+    # gt_config.cache_settings["root_path"] = custom_path
+    # p1 = OrchestratedProgam(backend, DaCeOrchestration.Python)
+    # p1()
+    # assert os.path.exists(
+    #     f"{custom_path}/.gt_cache_000000/py38_1013/gtcpu_ifirst/__main__/_stencil/"
+    # )
+
+    # # Check relocability by copying the second cache directory,
+    # # changing the path of gt_config.cache_settings and trying to Run on it
+    # relocated_path = f"{working_dir}/.my_relocated_cache_path"
+    # shutil.copytree(custom_path, relocated_path, dirs_exist_ok=True)
+    # gt_config.cache_settings["root_path"] = relocated_path
+    # p2 = OrchestratedProgam(backend, DaCeOrchestration.Python)
+    # p2()
+    # assert os.path.exists(
+    #     f"{relocated_path}/.gt_cache_000000/py38_1013/gtcpu_ifirst/__main__/_stencil/"
+    # )
```

Comment:
?

Reply:
Python `so` system doesn't know how to dereg memory properly. If you ask to open the so twice... it dies. DaCe as a way to rename the symbol to go around that but `gt` doesn't. I raised that with GT4Py team, it's considered minor for now but I wanted to keep the test in because relocatability is important.

Reply:
Is there a way to avoid merging it in commented like this?